### ISSUE TITLE
Make indeterminate progress small size by default

### ISSUE THREAD

**Issue by timothyjc** [JakeWharton/ActionBarSherlock#387]
Feature Request:

Currently the default indeterminate progress is standard size, whereas across all android platforms the default size of the indeterminate progressbar in the title window is small. And it is small for good reason - it looks much better and is similar size to the other icons.

I found this post on setting the size to small but it seems like it was left hanging...

https://groups.google.com/forum/?fromgroups#!topic/actionbarsherlock/JOnM4QP3fJc

TLDR: I would like to see the default indeterminate progressbar size small with an easy option for setting the size to medium.

Thanks!

PS. Great work Jake :)

**Comment by JakeWharton** [comment]
Sorry but the ICS action bar uses the large size therefore so does this library. The styles are available for your customization as that linked post describes. You should be able to create your own theme which uses the built-in small progress style or grab the ICS ones for a consistent look.

**Comment by timothyjc** [comment]
Ok... thanks for such fast responses.

**Comment by LorneLaliberte** [comment]
Where does ICS use such a large progress indicator, Jake? The one in Gmail's action bar in Jelly Bean is two thirds the size of the one in ActionBarSherlock.

I haven't done an exhaustive search, but so far the only large progress indicator I've spotted with this size is the one used in the main view of an activity (e.g. in a listview). I haven't noticed one in the action bar that was this large yet.

**Comment by SimonVT** [comment]
Open up the demos app, then go to "Indeterminate Progress". Also, gmail most likely uses an action view or something, and not the action bar's indeterminate progress feature.

**Comment by LorneLaliberte** [comment]
Yeah, that will show the default action bar implementation, but what I mean is that I've never actually seen such a large indeterminate progress indicator in the wild on ICS or Jelly Bean. I've only ever seen smaller ones (so far).

I guess the more proper question is why did Google make the action bar's indeterminate progress so big in the first place, if they didn't like the look of it that way either... :)

**Comment by imminent** [comment]
@LorneLaliberte, @SimonVT is correct. If you're looking to Google apps for the default action bar implementation of the indeterminate progress, it will probably be hard to find, they tend to (and with good reason) use a refresh button that they turn into an indeterminate progress indicator when "progressing", so the default action bar indeterminate progress indicator won't show up. Check out the source for iosched the 2012 Google Session app if you want to see what they do. It's kind of the Android pull-to-refresh pattern done correctly (aka, it works better than pull-to-refresh for Android).

**Comment by eladnava** [comment]
You could control the size of your intermediate progress bar by using an Action View in the onCreateOptionsMenu() like so:

```
    //----------------------------
    // Add refresh in Action Bar
    //----------------------------

    LoadingItem = OptionsMenu.add(Menu.NONE, Menu.NONE, Menu.NONE, "Loading");

    //----------------------------
    // Set up the view
    //----------------------------

    LoadingItem.setActionView(R.layout.loading);

    //----------------------------
    // Specify the show flags
    //----------------------------

    LoadingItem.setShowAsAction(MenuItem.SHOW_AS_ACTION_ALWAYS);
```

R.layout.loading:

```
    <?xml version="1.0" encoding="utf-8"?>
    <LinearLayout
            xmlns:android="http://schemas.android.com/apk/res/android"
            android:layout_width="wrap_content"
            android:layout_height="wrap_content">
        <ProgressBar
                android:layout_width="35dp"
                android:layout_height="35dp"
                android:layout_marginRight="5dp" />
    </LinearLayout>
```

You would then call LoadingItem.setVisible(); instead of setSupportProgressBarIndeterminateVisibility() from your activity's UI thread to show or hide the loading indicator.
